### PR TITLE
[WIP] Rewrite toast unit tests

### PR DESCRIPTION
**Pull request by bershanskiy**
@Johann-S Here are Toast unit tests.
* I'm still working on two tests (marked with TODO), I'll finish them tomorrow
* This time I didn't try to add any new unit tests

### CODE REVIEW

**Comment by Johann-S** [comment]
Just few notes:
 - Do not exclude `js/dist` in the .gitgnore
 - You have to do one `describe` per methods

When you're done with that just /CC me 😉 

I'll update my branch soon, so to keep up to date you can do:
`git pull upstream jo-jasmine --rebase` 

`upstream` is the remote which target Bootstrap

File: .gitignore [hunk 1]
```diff
@@ -46,3 +46,9 @@ Thumbs.db
 # Folders to ignore
 /coverage/
 /node_modules/
+
+# Convenience
+/dist/
```
Comment (XhmikosR) [inline .gitignore]: yeah all these need to be gone.

Comment (bershanskiy) [reply .gitignore]: Sorry about that. Fixed now.

**Comment by bershanskiy** [comment]
@Johann-S Thanks for your comments. Would you please take a look at this now?

I still have one TODO:
> TODO: There should be a better way to emulate jQuery .one

Currently I use a flag that that I set upon the first call, but I feel there should be a better way.

**Comment by bershanskiy** [comment]
@Johann-S I didn't see your recent force-push (17 min ago) and just (5 min ago) force-pushed after `git pull upstream jo-jasmine --rebase`. I didn't mean to override your changes, if needed I'll revert my force-push back.

**Comment by bershanskiy** [comment]
@Johann-S Thanks for your comments. I tried to fix all the issues and squashed all commits into one, I hope it's easier to review this way.

File: js/src/toast/toast.spec.js [hunk 2]
```diff
@@ -0,0 +1,222 @@
+import Toast from './toast'
+
+import EventHandler from '../dom/eventHandler'
```
Comment (Johann-S) [inline js/src/toast/toast.spec.js]: The separated lines isn't needed

File: js/src/toast/toast.spec.js [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+  beforeAll(() => {
+    fixtureEl = getFixture()
+  })
+
+  afterEach(() => {
+    clearFixture()
+  })
+
+  it('should return version', () => {
```
Comment (Johann-S) [inline js/src/toast/toast.spec.js]: `describe('VERSION')`

File: js/src/toast/toast.spec.js [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+          expect(toastEl.classList.contains('show')).toEqual(false)
+
+          done()
+        }, 1)
+      })
+      toast.show()
+    })
+  })
+
+  describe('config', () => {
```
Comment (Johann-S) [inline js/src/toast/toast.spec.js]: config isn't a public method, you should move those tests in `constructor` describe

File: js/src/toast/toast.spec.js [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+      toastEl.addEventListener('hidden.bs.toast', () => {
+        expect(toastEl.classList.contains('show')).toEqual(false)
+        done()
+      })
+
+      toast.show()
+    })
+  })
+
+  describe('constructor', () => {
```
Comment (Johann-S) [inline js/src/toast/toast.spec.js]: should be the first `describe`

File: js/src/toast/toast.spec.js [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+    it('should expose default setting to allow to override them', () => {
```
Comment (Johann-S) [inline js/src/toast/toast.spec.js]: should be relative to the `Default` describe

File: js/src/toast/toast.spec.js [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+      toastEl.addEventListener('shown.bs.toast', () => {
+        expect(toastEl.classList.contains('show')).toEqual(true)
+        done()
+      })
+
+      toast.show()
+    })
+  })
+
+  describe('event-key', () => {
```
Comment (Johann-S) [inline js/src/toast/toast.spec.js]: those tests should be in the `constructor` describe because it's here where the event listener is added

**Review by Johann-S — CHANGES_REQUESTED** [review]
It lacks of `DefaultType` and `Default` describe

**Comment by fullygit** [comment]
https://github.com/fullygit/fullygit

File: js/src/toast/toast.spec.js (hunk 6 above)
Comment (bershanskiy) [reply js/src/toast/toast.spec.js]: @Johann-S I'm not sure what you mean. Would you please elaborate?

Comment (Johann-S) [reply js/src/toast/toast.spec.js]: Default is a public method so it should be in a separate `describe`

Comment (bershanskiy) [reply js/src/toast/toast.spec.js]: @Johann-S I made Default and DefaultType into their own describes ([two days ago](https://github.com/twbs/bootstrap/commit/299ad44fc60fbbde2e5fb3572ceaa5a8b10dffaf), I just did not mention it in the comment).
I hope my contributions are actually helpful and not wasting your time.

Comment (Johann-S) [reply js/src/toast/toast.spec.js]: Your contributions are totally valuable for us thanks for that 👍 It's just I like to follow the same rules for the structure of our unit tests 👍

**Comment by Johann-S** [comment]
Hi @bershanskiy,

I fixes some lint error on your commit.

You have one method left to test:

image

And I'll be happy to merge your PR 👍 
Thanks again 👍

**Comment by Johann-S** [comment]
Hi @bershanskiy I cherry picked your commit in my branch, thanks for your work 👍

**Pull request closed without merging**